packages/ipfs-message-port-client/src/block.js

```diff
@@ -0,0 +1,154 @@
+'use strict'
+
+const { Client } = require('./client')
+const { encodeCID, decodeCID } = require('ipfs-message-port-protocol/src/cid')
+const { decodeError } = require('ipfs-message-port-protocol/src/error')
+const {
+  encodeBlock,
+  decodeBlock
+} = require('ipfs-message-port-protocol/src/block')
+
+/**
+ * @typedef {import('cids')} CID
+ * @typedef {import('ipfs-message-port-server/src/block').Block} Block
+ * @typedef {import('ipfs-message-port-server/src/block').EncodedBlock} EncodedBlock
+ * @typedef {import('ipfs-message-port-server/src/block').Rm} EncodedRmEntry
+ * @typedef {import('ipfs-message-port-server/src/block').BlockService} BlockService
+ * @typedef {import('./client').ClientTransport} Transport
+ */
+
+/**
+ * @class
+ * @extends {Client<BlockService>}
+ */
+class BlockClient extends Client {
+  /**
+   * @param {Transport} transport
+   */
+  constructor (transport) {
+    super('block', ['put', 'get', 'rm', 'stat'], transport)
+  }
+
+  /**
+   * Get a raw IPFS block.
+   * @param {CID} cid - A CID that corresponds to the desired block
+   * @param {Object} [options]
+   * @param {number} [options.timeout] - A timeout in ms
+   * @param {AbortSignal} [options.signal] - Can be used to cancel any long
+   * running requests started as a result of this call
+   * @param {Transferable[]} [options.transfer] - References to transfer to the
+   * worker if passed.
+   * @returns {Promise<Block>}
+   */
+  async get (cid, options = {}) {
+    const { transfer } = options
+    const { block } = await this.remote.get({
+      ...options,
+      cid: encodeCID(cid, transfer)
```

Comment:
I still need `encodeCID` for sending stuff from the server side. There we know everything can be transferred. And since I do have `encodeCID / decodeCID` it's actually easier to just use them across the board than use different encoding / decoding options.